reV/SAM/generation.py

```diff
@@ -1149,6 +1161,221 @@ def default():
+                   .format(self.sam_sys_inputs["resource_potential"],
+                           gross_gen))
+            logger.warning(msg)
+            warn(msg)
+
+        logger.debug("Setting the resource potential to {}".format(gross_gen))
+        self.sam_sys_inputs["resource_potential"] = gross_gen
+        self["ui_calculations_only"] = 0
+
+    def _create_pysam_wfile(self, resource, meta):
+        """Create PySAM weather input file.
+
+        Geothermal module requires a weather file, but does not actually
+        require any weather data to run. Therefore, an empty file is
+        generated and passed through.
+
+        Parameters
+        ----------
+        resource : pd.DataFrame
+            Time series resource data for a single location with a
+            pandas DatetimeIndex. There must be columns for all the
+            required variables to run the respective SAM simulation.
+        meta : pd.DataFrame | pd.Series
+            Meta data corresponding to the resource input for the single
+            location. Should include values for latitude, longitude,
+            and timezone.
+
+        Returns
+        -------
+        fname : str
+            Name of weather csv file.
+
+        Notes
+        -----
+        PySAM will not accept data on Feb 29th. For leap years,
+        December 31st is dropped and time steps are shifted to relabel
+        Feb 29th as March 1st, March 1st as March 2nd, etc.
+        """
+        # pylint: disable=attribute-defined-outside-init, consider-using-with
+        self._temp_dir = TemporaryDirectory()
+        fname = os.path.join(self._temp_dir.name, 'weather.csv')
+        logger.debug('Creating PySAM weather data file: {}'.format(fname))
+
+        # ------- Process metadata
+        m = pd.DataFrame(meta).T
+        m = m.rename({"latitude": "Latitude", "longitude": "Longitude",
+                      "timezone": "Time Zone"}, axis=1)
+
+        m[["Latitude", "Longitude", "Time Zone"]].to_csv(fname, index=False,
+                                                         mode='w')
+
+        # --------- Process data, blank for geothermal
+        time_index = resource.index
+        mask = (time_index.month == 2) & (time_index.day == 29)
+        time_index = time_index[~mask]
+
+        df = pd.DataFrame(index=time_index)
+        df['Year'] = time_index.year
+        df['Month'] = time_index.month
+        df['Day'] = time_index.day
+        df['Hour'] = time_index.hour
+        df['Minute'] = time_index.minute
+        df.to_csv(fname, index=False, mode='a')
+
+        return fname
+
+    def run_gen_and_econ(self):
+        """Run SAM generation and possibility follow-on econ analysis."""
+        try:
+            super().run_gen_and_econ()
+        except SAMExecutionError as e:
+            logger.error("Skipping site {}; received sam error: {}"
+                         .format(self._site, str(e)))
+            self.outputs = {}
```

Comment:
Would we want default outputs? e.g. capacity factor = 0? What outputs get written if the sim fails?

Reply:
Yea the reV `Gen` class initializes all outputs to 0, so those are the default outputs. If the sim fails, the default initialized value is *not* overwritten, so the outputs are all 0 for that point.

Happy to walk through this in finer detail during a short call if you'd like!